### PR TITLE
[docs] Add capability to add caption under an image

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-10260

# How

- Updated the ImageSpotlight component to add the capability of adding a `caption` under any image. Right now, this prop is optional. We can also use `alt` instead of `caption` here (just a suggestion). Instead of making too many changes in our docs for adding caption, I wanted to make sure the modifications in the PR are acceptable in terms of design.

# Test Plan

- Run docs locally
- Go to a doc that uses `ImageSpotlight` and a `caption` prop to it. Value of this prop is a meaningful string that describes the contents of the image.

**Preview**

Here is an example of how adding caption to the `ImageSpotlight` will look. For this example, I used [Account Types](http://localhost:3002/accounts/account-types/) since that guide has a significant amount of images.

CleanShot 2023-09-29 at 22 40 48@2x

CleanShot 2023-09-29 at 22 41 14@2x

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/components/plugins/ImageSpotlight.tsx [hunk 1]
```diff
@@ -6,21 +6,36 @@ type Props = {
...
+        style={{
+          textAlign: 'center',
+          backgroundColor: theme.background.subtle,
+          paddingTop: 10,
+          paddingBottom: 10,
+          marginTop: 20,
+          marginBottom: 20,
+          ...containerStyle,
+        }}>
+        <img src={src} alt={alt} style={style} className="inline" />
```
Comment (Simek) [inline docs/components/plugins/ImageSpotlight.tsx:25]: Can we move caption inside the div with padding and relay only on top margin for the spacing?

File: docs/components/plugins/ImageSpotlight.tsx [hunk 2]
```diff
@@ -6,21 +6,36 @@ type Props = {
...
-        paddingTop: 10,
-        paddingBottom: 10,
-        marginTop: 20,
-        marginBottom: 20,
-        ...containerStyle,
-      }}>
-      <img src={src} alt={alt} style={style} className="inline" />
-    </div>
+    <figure>
+      <div
```
Comment (Simek) [inline docs/components/plugins/ImageSpotlight.tsx:15]: With change above we can replace `div` with `figure` directly.

```suggestion
    <figure
```

**Comment by amandeepmittal** [comment]
@Simek, I've incorporated the changes from your review.

Here is the end result:

 
CleanShot 2023-10-01 at 11 28 20@2x

CleanShot 2023-10-01 at 11 28 11@2x

**Review by Simek — APPROVED** [review]
Thank you for the updates, LGTM! 🚀

**Pull request merged**